packages/react-dom/src/__tests__/ReactBrowserEventEmitter-test.internal.js

```diff
@@ -9,360 +9,480 @@
 
 'use strict';
 
-let EventPluginHub;
-let EventPluginRegistry;
 let React;
 let ReactDOM;
-let ReactDOMComponentTree;
-let ReactBrowserEventEmitter;
 let ReactTestUtils;
-
-let idCallOrder;
-const recordID = function(id) {
-  idCallOrder.push(id);
-};
-const recordIDAndStopPropagation = function(id, event) {
-  recordID(id);
-  event.stopPropagation();
-};
-const recordIDAndReturnFalse = function(id, event) {
-  recordID(id);
-  return false;
-};
-const LISTENER = jest.fn();
-const ON_CLICK_KEY = 'onClick';
-const ON_CHANGE_KEY = 'onChange';
-const ON_MOUSE_ENTER_KEY = 'onMouseEnter';
-
-let GRANDPARENT;
-let PARENT;
-let CHILD;
-
-let getListener;
-let putListener;
-let deleteAllListeners;
+let LISTENER;
 
 let container;
 
-function registerSimpleTestHandler() {
-  putListener(CHILD, ON_CLICK_KEY, LISTENER);
-  const listener = getListener(CHILD, ON_CLICK_KEY);
-  expect(listener).toEqual(LISTENER);
-  return getListener(CHILD, ON_CLICK_KEY);
-}
+let createInitialInstance;
+
+let updateInstance;
+
+let childRef;
 
 describe('ReactBrowserEventEmitter', () => {
   beforeEach(() => {
     jest.resetModules();
-    LISTENER.mockClear();
+    LISTENER = jest.fn();
 
-    // TODO: can we express this test with only public API?
-    EventPluginHub = require('events/EventPluginHub');
-    EventPluginRegistry = require('events/EventPluginRegistry');
     React = require('react');
     ReactDOM = require('react-dom');
-    ReactDOMComponentTree = require('../client/ReactDOMComponentTree');
-    ReactBrowserEventEmitter = require('../events/ReactBrowserEventEmitter');
     ReactTestUtils = require('react-dom/test-utils');
 
     container = document.createElement('div');
     document.body.appendChild(container);
 
-    let GRANDPARENT_PROPS = {};
-    let PARENT_PROPS = {};
-    let CHILD_PROPS = {};
-
-    function Child(props) {
-      return <div ref={c => (CHILD = c)} {...props} />;
+    class Child extends React.PureComponent {
```

Comment:
Have you found out why this PureComponent is needed? It was also set on the wrapper of Child previously - Maybe to avoid the Child from receiving updates?